app/src/components/authentication/SignUpForm.tsx

```diff
@@ -16,44 +16,46 @@ import CloseIcon from '@mui/icons-material/Close';
 import {styled} from '@mui/system';
 import GoogleIcon from '@mui/icons-material/Google';
 import {useFormik} from 'formik';
-import {object, string} from 'yup';
 import {SignInRequest} from '../../services/auth';
+import {PasswordValidation} from './PasswordValidation';
+import pwValidate, {validationSchema} from '../common/PasswordValidationSchema';
 
+// QUESTIONS TO ASK NEXT MEETING - 04/11/23:
+// 1. updating a step behind
 interface SignUpFormProps {
   onSubmit: ({email, password}: SignInRequest) => Promise<void>;
   setErrorMessage: React.Dispatch<React.SetStateAction<string>>;
   errorMessage: string;
 }
 
-const validationSchema = object({
-  email: string().email().required('email is required'),
-  password: string()
-    .required('password is required')
-    .min(8, 'password must be at least 8 characters')
-    .matches(/^(?=.*[0-9])/, 'password must contain at least one number')
-    .matches(/^(?=.*[a-z])/, 'Must contain at least one lowercase character')
-    .matches(/^(?=.*[A-Z])/, 'Must contain at least one uppercase character')
-    .matches(
-      /^(?=.*[!@#%&])/,
-      'password must contain at least one special character',
-    ),
-});
-
 export const SignUpForm = ({
   onSubmit,
   errorMessage,
   setErrorMessage,
 }: SignUpFormProps) => {
-  const {handleSubmit, handleChange, values, touched, errors} = useFormik({
-    initialValues: {
-      email: '',
-      password: '',
-    },
-    validationSchema,
-    onSubmit: values => {
-      onSubmit(values);
-    },
-  });
+  const {handleSubmit, handleChange, values, touched, errors, isValid} =
+    useFormik({
+      initialValues: {
+        email: '',
+        password: '',
+      },
+      validationSchema: validationSchema,
+
+      onSubmit: values => {
+        onSubmit(values);
+      },
+    });
+
+  // set errors to "contains" state, passdown to password validation as prop
+  // and dynamically change UI according to what is in errors array
+
+  const [errorsLeft, setErrorsLeft] = useState<string[]>([]);
```

Comment:
I'm wondering if there is a way we could encapsulate this state and the validation logic inside the PasswordValidation component since we have to use this component in multiple places. Maybe if we just pass the password from Formik down and handle the validation and state update in a useEffect?